### PR TITLE
Port examples to use Create React App

### PR DESCRIPTION
**Pull request by iswanj**
If this is ok with you, Then I'll port all examples to use [Create React App](https://github.com/facebookincubator/create-react-app)

Reference to [this issue](https://github.com/facebookincubator/create-react-app/issues/737)

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: examples/hello-world/README.md [hunk 1]
```diff
@@ -3,26 +3,13 @@ React Intl Hello World Example
 
 This is a very simple — yet runnable app — showing how to use React Intl to format message (string) that contains a placeholder, plural, and number.
 
-## Running
+**This project template was built with [Create React App](https://github.com/facebookincubator/create-react-app).**
```
Comment (gaearon) [inline examples/hello-world/README.md]: This looks too much like advertising here 😄 
I think we should be more humble and just link to the [user guide](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md) instead.

File: examples/hello-world/index.html [hunk 2]
```diff
@@ -1,14 +0,0 @@
-<!DOCTYPE html>
-<html lang="en">
-<head>
-    <meta charset="utf-8">
-    <title>React Intl Hello World Example</title>
-</head>
-<body>
-    <div id="container"></div>
-    <script src="https://cdn.polyfill.io/v1/polyfill.min.js?features=Intl.~locale.en"></script>
```
Comment (gaearon) [inline examples/hello-world/index.html]: We need to keep this script in `index.html`.

File: examples/hello-world/public/index.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tag above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favico.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (gaearon) [inline examples/hello-world/public/index.html]: Please use the `title` from original HTML file.

File: examples/hello-world/public/index.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
```
Comment (gaearon) [inline examples/hello-world/public/index.html]: Both favicon and the comment below can be completely removed, they are irrelevant to this project.

File: examples/hello-world/src/App.test.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
```
Comment (gaearon) [inline examples/hello-world/src/App.test.js]: There is no need for this file in this example, as it doesn’t have any tests.

File: examples/hello-world/src/logo.svg [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3">
```
Comment (gaearon) [inline examples/hello-world/src/logo.svg]: There is no need for this file in this example.

File: examples/hello-world/package.json [hunk 7]
```diff
@@ -4,19 +4,20 @@
...
-    "react-dom": "global:ReactDOM"
+  "dependencies": {
+    "react": "^15.3.2",
+    "react-dom": "^15.3.2",
+    "react-intl": "^2.1.5"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test --env=jsdom",
```
Comment (gaearon) [inline examples/hello-world/package.json]: We can remove the `test` command here because it is irrelevant to this example.
Perhaps `eject` too.

**Comment by ericf** [comment]
@iswanj cool, thanks! I have this same example converted over to `create-react-app` in a WIP state on my other computer too. I will sync this with some other changes I have to `npm link` all the things so the examples can run against the current working version of the lib.

**Comment by gaearon** [comment]
Thank you for merging!

**Pull request merged**